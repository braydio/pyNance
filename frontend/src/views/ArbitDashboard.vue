<template>
  <div class="container py-4">
    <h1>Arbit Dashboard</h1>
    <ArbitAlerts />
  </div>
</template>

<script setup>
/**
 * Displays metrics for the optional Arbit dashboard.
 */
import ArbitAlerts from '@/components/ArbitAlerts.vue'
</script>
