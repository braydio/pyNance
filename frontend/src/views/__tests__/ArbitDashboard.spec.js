// @vitest-environment jsdom
import { describe, it, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import ArbitDashboard from '../ArbitDashboard.vue'

const PassThrough = { template: '<div class="pass-through"><slot /></div>' }

const PageHeaderStub = {
  template: '<div class="page-header-stub"><slot name="title" /><slot name="subtitle" /></div>',
}

const createStub = (testId) => ({
  template: `<div data-test="${testId}"></div>`,
})

describe('ArbitDashboard.vue', () => {
  const mountView = () =>
    shallowMount(ArbitDashboard, {
      global: {
        stubs: {
          BasePageLayout: PassThrough,
          PageHeader: PageHeaderStub,
          Card: PassThrough,
          ArbitStatus: createStub('arbit-status-stub'),
          ArbitControls: createStub('arbit-controls-stub'),
          ArbitMetrics: createStub('arbit-metrics-stub'),
          ArbitAlerts: createStub('arbit-alerts-stub'),
          ArbitOpportunities: createStub('arbit-opportunities-stub'),
          ArbitTrades: createStub('arbit-trades-stub'),
        },
      },
    })

  it('renders dashboard sections with expected components', () => {
    const wrapper = mountView()

    const statusSection = wrapper.find('[data-testid="status-controls"]')
    const metricsSection = wrapper.find('[data-testid="metrics-alerts"]')
    const tradesSection = wrapper.find('[data-testid="opportunities-trades"]')

    expect(statusSection.exists()).toBe(true)
    expect(metricsSection.exists()).toBe(true)
    expect(tradesSection.exists()).toBe(true)

    expect(statusSection.find('[data-test="arbit-status-stub"]').exists()).toBe(true)
    expect(statusSection.find('[data-test="arbit-controls-stub"]').exists()).toBe(true)
    expect(metricsSection.find('[data-test="arbit-metrics-stub"]').exists()).toBe(true)
    expect(metricsSection.find('[data-test="arbit-alerts-stub"]').exists()).toBe(true)
    expect(tradesSection.find('[data-test="arbit-opportunities-stub"]').exists()).toBe(true)
    expect(tradesSection.find('[data-test="arbit-trades-stub"]').exists()).toBe(true)
  })

  it('matches snapshot', () => {
    const wrapper = mountView()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
