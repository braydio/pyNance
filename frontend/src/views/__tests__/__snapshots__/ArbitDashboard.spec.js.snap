// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArbitDashboard.vue > matches snapshot 1`] = `
"<div data-v-db7eac05="" class="pass-through arbit-dashboard" gap="gap-10">
  <div data-v-db7eac05="" class="page-header-stub" icon="(props, { slots, attrs }) => vue.h(
  Icon,
  {
    ...attrs,
    ...props,
    iconNode,
    name: iconName
  },
  slots
)">Arbitrage Command CenterMonitor engine status, opportunities, and trades</div>
  <section data-v-db7eac05="" data-testid="status-controls" class="grid grid-cols-1 gap-6 xl:grid-cols-3">
    <div data-v-db7eac05="" class="pass-through h-full p-6 space-y-4">
      <h2 data-v-db7eac05="" class="section-title">Engine Status</h2>
      <div data-v-db7eac05="" data-test="arbit-status-stub"></div>
    </div>
    <div data-v-db7eac05="" class="pass-through h-full p-6 space-y-4 xl:col-span-2">
      <h2 data-v-db7eac05="" class="section-title">Control Panel</h2>
      <div data-v-db7eac05="" data-test="arbit-controls-stub"></div>
    </div>
  </section>
  <section data-v-db7eac05="" data-testid="metrics-alerts" class="grid grid-cols-1 gap-6 lg:grid-cols-3">
    <div data-v-db7eac05="" class="pass-through h-full p-6 space-y-4 lg:col-span-2">
      <h2 data-v-db7eac05="" class="section-title">Performance Metrics</h2>
      <div data-v-db7eac05="" data-test="arbit-metrics-stub"></div>
    </div>
    <div data-v-db7eac05="" class="pass-through h-full p-6 space-y-4">
      <h2 data-v-db7eac05="" class="section-title">Live Alerts</h2>
      <div data-v-db7eac05="" data-test="arbit-alerts-stub"></div>
    </div>
  </section>
  <section data-v-db7eac05="" data-testid="opportunities-trades" class="grid grid-cols-1 gap-6 xl:grid-cols-3">
    <div data-v-db7eac05="" class="pass-through h-full p-6 space-y-4">
      <h2 data-v-db7eac05="" class="section-title">Current Opportunities</h2>
      <div data-v-db7eac05="" data-test="arbit-opportunities-stub"></div>
    </div>
    <div data-v-db7eac05="" class="pass-through h-full p-6 space-y-4 xl:col-span-2 overflow-x-auto">
      <h2 data-v-db7eac05="" class="section-title">Recent Trades</h2>
      <div data-v-db7eac05="" data-test="arbit-trades-stub"></div>
    </div>
  </section>
</div>"
`;
